### ISSUE TITLE
[Question] Update Traits properties on a plugin

### ISSUE THREAD

**Issue by Lerdouille** [artf/grapesjs#2730]
Hi,

is it possible to change easily some properties in the Traits configuration of a plugin ?
I"m trying to translate some strings in the grapesjs-echarts plugin, and i kindy stucked after reading your documentation and looking for the functions to do so...

Could please tell me if it's possible to change some options plugins when it's not implemented by default ?

Regards

**Comment by artf** [comment]
You should be able to do it via i18n module
```js
editor.I18n.addMessages({
        en: {
          traitManager: {
            labels: {
				title: 'My Title', // `title` is the name of the trait
			}
          },
        }
      });
```

**Comment by Lerdouille** [comment]
Hi,

thank you for the answer. I agree with you, it changes indeed the default labels on Traits (.

But what about a plugin, or a component not added in the i18n module ? 
For example, the type "map" include in your initial javascript, how can i update without override your plugin it's properties defined below :
type: 'map',
    src: '',
    void: 0,
    mapUrl: 'https://maps.google.com/maps',
    tagName: 'iframe',
    mapType: 'q',
    address: '',
    zoom: '1',
    attributes: {
      frameborder: 0
    },
    toolbar: _Component__WEBPACK_IMPORTED_MODULE_2__["default"].prototype.defaults.toolbar,
    traits: [{
      label: 'Address',
      name: 'address',
      placeholder: 'eg. London, UK',
      changeProp: 1
    }, {
      type: 'select',
      label: 'Map type',
      name: 'mapType',
      changeProp: 1,
      options: [{
        value: 'q',
        name: 'Roadmap'
      }, {
        value: 'w',
        name: 'Satellite'
      }]
    }, {
      label: 'Zoom',
      name: 'zoom',
      type: 'range',
      min: '1',
      max: '20',
      changeProp: 1
    }]

Basically, i would like to update some properties for any type of component after the editor initialization. I know there is a function editor.DomComponents.addType to implement a new type or completely replace the component with it's ID. 
Is there a editor.DomComponents.updateType where i could juste change label name ?

Regards

**Comment by Lerdouille** [comment]
Hi again,

any idea to help me to update traits default properties after the plugin initialization ?
In my previous sampe, i have a hard coded type 'map' in my grapes.js , i would like to update some traits properties in my html page with some translation and custom values.

Best regards

**Comment by Lerdouille** [comment]
For the record, i'm not in a Node.Js environment

**Comment by Lerdouille** [comment]
Ok so basically i succeed to update traits properties by doing this after the plugin initialization :

editor.DomComponents.getType('map).model.prototype.defaults.traits[0].label = "New Label";

But i can't figure it out how to update traits properties when there is a listener on them...
For example, reguarding the 'video' type, when there is a change on the source select, all the traits are changed with functions :
- getYoutubeTraits
- getVimeoTraits
- getSourceTraits

So i can't use my trick to translate for example the string "Provider"....

Any idea how i can do that ?

Best regards

**Comment by artf** [comment]
I don't understand what is your issue, you can still use the i18n module (even if the trait is added externally from a plugin)
```js
editor.I18n.addMessages({
        en: {
          traitManager: {
            labels: {
				provider: 'Provider',
			}
          },
        }
      });
```
